### PR TITLE
Add pytest and profiling, switch from setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by peijin94**
related to [https://github.com/ovro-eovsa/ovro-lwa-solar/issues/86](https://github.com/ovro-eovsa/ovro-lwa-solar/issues/86)

I left the decorator `@profile` on some of the functions for future profiling. 
The decorator `@profile`  will only be activated when the script is running with the command `kernprof myscritp.py`

To run the profiler,

```
# active the env
# pip install line_profiler
python -m kernprof -lvr profiling.py
```

To run the test:

```
# pip install pytest
pytest tests/test_modules.py
```

### CODE REVIEW

File: tests/test_modules.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # copy caltables and ms files to the tmp directory
+    os.chdir(dir_tmp)
+    data_tar = '/data07/peijinz/testdir/testdata.tar'
+    os.system('cp '+data_tar+' ./')
+    os.system('tar -xvf testdata.tar')
+    yield dir_tmp
+
+@pytest.fixture
+def run_image_quick(prepare_data_local):
+    dir_tmp = prepare_data_local
```
Comment (surajit4444mondal) [inline tests/test_modules.py:36]: prepare_data_local uses your area on calim7. Do we want to use prepare_data instead?

**Comment by surajit4444mondal** [comment]
Is it possible to use the logging level to determine whether we want wsclean to run quietly? Then we do not need to pass "quiet" as an argument to do_selfcal.

Also can you please also add in the issue how to run the profiler?

File: tests/test_modules.py (hunk 1 above)
Comment (peijin94) [reply tests/test_modules.py:36]: Yeah, for general tests not on calim07, one should use prepare_data

**Comment by peijin94** [comment]
> Is it possible to use the logging level to determine whether we want wsclean to run quietly? Then we do not need to pass "quiet" as an argument to do_selfcal.
> 
> Also can you please also add in the issue how to run the profiler?

PR description updated

**Review by surajit4444mondal — APPROVED** [review]

**Review by binchensun — APPROVED** [review]

**Pull request merged**